feat(express): migrate signpayload to typed routes

danielzhao122 opened the pull request:
Typed router for signPayload
Added UT: "express.ofc.signPayload" in test/unit/typedRoutes/decode.ts
TICKET: WP-5443

danielzhao122 committed:
feat(express): migrate signpayload to typed routes

TICKET: WP-5443

zahin-mohammad commented on modules/express/src/typedRoutes/api/v2/ofcSignPayload.ts:
```diff
@@ -0,0 +1,31 @@
+  method: 'POST',
+  request: httpRequest({ body: OfcSignPayloadBody }),
+  response: {
+    200: t.type({ payload: t.string, signature: t.string }),
```
Please define the response codec as a variable and add some jsdoc's regarding their fields. 

The codecs need to be their own variables due to how the code generation works.

zahin-mohammad replied:
Also to ensure this works as we expect it to, lets find an existing test for `signPayload` and use the codec to decode the response

zahin-mohammad commented on modules/express/src/typedRoutes/api/v2/ofcSignPayload.ts:
```diff
@@ -0,0 +1,31 @@
+ * POST /api/v2/ofc/signPayload
+ */
+export const OfcSignPayloadBody = {
+  walletId: NonEmptyString,
```
Missing jsdocs. In this case, since there is no platform.yaml to reference, lets use the jsdocs from the underlying code.

zahin-mohammad commented on modules/express/src/clientRoutes.ts:
```diff
@@ -1649,8 +1642,8 @@ export function setupAPIRoutes(app: express.Application, config: Config): void {
 
-  // sign arbitrary payloads w/ trading account key
-  app.post(`/api/v2/ofc/signPayload`, parseBody, prepareBitGo(config), promiseWrapper(handleV2OFCSignPayload));
+  // sign arbitrary payloads w/ trading account key (migrated to typed route express.ofc.signPayload)
+  // app.post(`/api/v2/ofc/signPayload`, parseBody, prepareBitGo(config), promiseWrapper(handleV2OFCSignPayload));
```
```suggestion
```

zahin-mohammad reviewed (changes requested):
Looks like there is some CI errors as well

danielzhao122 committed:
refactor: code

Added UT to test/unit/clientRoutes/signPayload.tsAlso it seems like this migration makes the UT for `Sign an arbitrary payload with trading account key` (ticket: https://bitgoinc.atlassian.net/browse/GO-1015) to pass.

TICKET: WP-5443

danielzhao122 committed:
refactor: added dependency io-ts-type to package.json

TICKET: WP-5443

danielzhao122 committed:
refactor: added dependency io-ts-type to package.json

TICKET: WP-5443

asset-metadata-bot[bot] committed:
feat: add new tokens for WIN-7171

danielzhao122 committed:
feat(express): migrate keychainlocal to typed routes

TICKET: WP-5414

danielzhao122 committed:
Merge branch 'master' into WP-5414-express-migrate-api-v2-coin-keychain-local-to-typed-routes

zahin-mohammad commented on modules/express/package.json:
```diff
@@ -51,6 +51,7 @@
     "dotenv": "^16.0.0",
     "express": "4.21.2",
     "io-ts": "npm:@bitgo-forks/io-ts@2.1.4",
+    "io-ts-types": "^0.5.19",
```
did you re-generate the lock file?

zahin-mohammad commented on modules/express/src/typedRoutes/api/v2/ofcSignPayload.ts:
```diff
@@ -0,0 +1,41 @@
+
+/**
+ * Sign an arbitrary payload using an OFC trading account key.
+ * @operationId express.ofc.signPayload
```
The operation id jsdoc should be part of the docs for `PostOfcSignPayload`

danielzhao122 committed:
refactor: added docs and updated req type

TICKET: WP-5414

parvaparikh-30 committed:
fix(express): signPayload API to handle stringified payload as req

Ticket: GNA-2162

parvbitgo067 committed:
feat: verify psbt withdraw lightning

TICKET: BTC-2470

danielzhao122 committed:
refactor: added express unit test for keychainLocal

TICKET: WP-5414

Venkat-Annavazzala committed:
chore: added new tokens for baseeth, lineath, seievm and flow

TICKET: WIN-7285

alextse-bg committed:
feat: add advanced wallet type

Ticket: WP-6028

danielzhao122 committed:
Merge branch 'master' into WP-5414-express-migrate-api-v2-coin-keychain-local-to-typed-routes

louib committed:
feat: configure learn to skip git operations

Ticket: VL-3486

yanxue-22 committed:
fix: replaced vulnerable request package by using axios

TICKET: DX-1558

danielzhao122 committed:
refactor: fixed merge conflict error

TICKET: WP-5414

zahin-mohammad committed:
fix: prepare-release should work with latest preid

This commit updates the prepare-release script to work with
the `@bitgo` stable relase org as well as the latest preid.

The script will use conventional commits for a specific module to
generate the next increment to use for the next version.

TICKET: WP-6047

krupanand-bitgo committed:
feat(sdk-coin-sol): onboarding sol 2022 token with transfer hook
onboarding new sol 2022 standard token with transfer hook extension in tesnet
Ticket: WIN-7258

nayandas190 committed:
feat: gate tokens for BitGo Singapore

TICKET: WIN-6923

manas-at-bitgo committed:
fix: modify codeowners for ams token export files

TICKET: WIN-7317

TICKET: WIN-7317

parasgarg-bitgo committed:
feat: adding iota coin construction in coin factory

Ticket: WIN-6873

TICKET: WIN-6873

Venkat-Annavazzala committed:
chore: update regex for raw trx hex validation

TICKET: WIN-7341

parvbitgo067 committed:
test: withdraw lightning psbt verification tests

TICKET: BTC-2470

parvbitgo067 committed:
feat: added bip174 to abstract lightning

TICKET: BTC-2470

danielzhao122 committed:
refactor: added decoding to unit tests and addressed missing jsdocs

TICKET: WP-5443

danielzhao122 committed:
Merge branch 'master' into WP-5443-express-migrate-api-v2-ofc-signPayload-to-typed-routes

TICKET: WP-5443

davidkaplanbitgo committed:
Merge pull request #7028 from BitGo/BTC-2470-verify-change-address

feat: verify psbt withdraw lightning

therealdwright committed:
Merge pull request #7060

parvaparikh-30 committed:
Merge pull request #7024 from BitGo/GNA-2162

fix(express): signPayload API to handle stringified payload as req

yanxue-22 committed:
Merge pull request #7055 from BitGo/DX-1558-Replace-Request-With-Axios

Dx 1558 replace vulnerable request package with axios

MohammedRyaan786 committed:
feat(sdk-coin-ton): add jetton transaction support
TICKET: COIN-5626

krupanand-bitgo committed:
Merge pull request #7059 from BitGo/WIN-7258-sol-2022-token-with-transfer-hook

feat(sol-coin-sol): onboarding sol 2022 token with transfer hook

sasikumar-bitgo committed:
fix(sdk-core): backup keychain creation to use correct privateMaterial

TICKET: WP-6057
Replaces the use of userPrivateMaterial with backupPrivateMaterial when adding the backup keychain in EcdsaMPCv2Utils. This ensures the correct private material is used for backup keychain generation.

Venkat-Annavazzala committed:
Merge pull request #7038 from BitGo/WIN-7285

chore: added new tokens for baseeth, lineath, seievm and flow

Ranjna-G committed:
feat(sdk-coin-ada): token build support

Ticket: WIN-7353

parasgarg-bitgo committed:
Merge pull request #7061 from BitGo/WIN-6873

feat: adding iota coin construction in coin factory

lokesh-bitgo committed:
feat(express): migrate calculateminerfeeinfo to typed routes

Ticket: WP-5400

Chinmay-Sharma1 committed:
feat(statics): onboard new bsc token

Ticket: COIN-5732

TICKET: COIN-5732

lokesh-bitgo committed:
Merge pull request #7071 from BitGo/WP-5400-express-migrate-api-calculateminerfeeinfo-to-typed-routes

feat(express): migrate calculateminerfeeinfo to typed routes

Venkat-Annavazzala committed:
Merge pull request #7063 from BitGo/WIN-7341

update regex for raw trx hex validation

Chinmay-Sharma1 committed:
Merge pull request #7072 from BitGo/COIN-5732-v1

feat(statics): onboard new bsc token

manas-at-bitgo committed:
Merge pull request #6953 from BitGo/ams-bot-tokens

feat: PR from asset-metadata-service for token onboardings

MohammedRyaan786 committed:
Merge pull request #7067 from BitGo/COIN-5626

feat(sdk-coin-ton): add jetton transaction support

zeeshanamjad-eng committed:
feat: replace asset id with fungible asset address

Ticket: TMS-1274

TICKET: TMS-1274

Ranjna-G committed:
Merge pull request #7070 from BitGo/WIN-7353-ada-token-build-support

feat(sdk-coin-ada): token build support

zeeshanamjad-eng committed:
Merge pull request #7073 from BitGo/TMS-1274

feat: replace apt usd1 asset id with fungible asset address

rohitsaw115 committed:
chore: adding evm_compatible_ims feature to tstt

ticket: win-7359

rohitsaw115 committed:
Merge pull request #7076 from BitGo/WIN-7359

chore: Adding EVM_COMPATIBLE_IMS feature to TSTT

nayandas190 committed:
Merge pull request #7057 from BitGo/win-6923

feat: gate tokens for BitGo Singapore

zahin-mohammad committed:
Merge pull request #7049 from BitGo/configure-lerna-to-skip-git-operations

feat: update release workflow

sasikumar-bitgo committed:
Merge pull request #7069 from BitGo/fix/WP-6057

fix(sdk-core): backup keychain creation to use correct privateMaterial for EcDSA MPCV2 wallets

danielzhao122 committed:
Merge pull request #6991 from BitGo/WP-5414-express-migrate-api-v2-coin-keychain-local-to-typed-routes

feat(express): migrate keychainlocal to typed routes

alextse-bg committed:
Merge pull request #7040 from BitGo/WP-6028

feat: migrate advanced wallet type from subtype to type

sachushaji committed:
feat: replace apt usd1 asset id with FA address

BREAKING CHANGE: FA address gets indexed for
transfers instead of move module address

TICKET: TMS-1274

sachushaji committed:
Merge pull request #7074 from BitGo/TMS-1274

feat!: replace apt usd1 asset id with Fungible asset address

zahin-mohammad reviewed (commented):
merge conflicts. re-request a review when ready

danielzhao122 committed:
feat(express): migrate signpayload to typed routes

TICKET: WP-5443

danielzhao122 committed:
refactor: code

Added UT to test/unit/clientRoutes/signPayload.tsAlso it seems like this migration makes the UT for `Sign an arbitrary payload with trading account key` (ticket: https://bitgoinc.atlassian.net/browse/GO-1015) to pass.

TICKET: WP-5443

danielzhao122 committed:
refactor: added decoding to unit tests and addressed missing jsdocs

TICKET: WP-5443

danielzhao122 committed:
Merge branch 'WP-5443-express-migrate-api-v2-ofc-signPayload-to-typed-routes' of github.com:BitGo/BitGoJS into WP-5443-express-migrate-api-v2-ofc-signPayload-to-typed-routes

TICKET: WP-5443
